src/System.Management.Automation/help/HelpCommands.cs

```diff
@@ -746,18 +746,13 @@ private static bool DoesCurrentRunspaceIncludeCoreHelpCmdlet()
                 IEnumerable<SessionStateCommandEntry> publicGetHelpEntries = iss
                     .Commands["Get-Help"]
                     .Where(entry => entry.Visibility == SessionStateEntryVisibility.Public);
-                if (publicGetHelpEntries.Count() != 1)
-                {
-                    return false;
-                }
 
-                foreach (SessionStateCommandEntry getHelpEntry in publicGetHelpEntries)
+                using (var en = publicGetHelpEntries.GetEnumerator())
                 {
-                    SessionStateCmdletEntry getHelpCmdlet = getHelpEntry as SessionStateCmdletEntry;
-                    if ((getHelpCmdlet != null) && (getHelpCmdlet.ImplementingType.Equals(typeof(GetHelpCommand))))
-                    {
-                        return true;
-                    }
+                    // Returns false if the number of elements is not 1
+                    return en.MoveNext()
+                        && ((en.Current as SessionStateCmdletEntry)?.ImplementingType.Equals(typeof(GetHelpCommand)) == true)
+                        && !en.MoveNext();
```

Comment:
This change is more complex and less readable. We should avoid direct use of `IEnumerator` where possible

Reply:
Yes, looks like a tricky code. @xtqqczze Where do you find the pattern?

Reply:
@iSazonov This is my own design, it avoids the use of `foreach` just to get the first item and avoids enumerating the whole collection with the `Count` method.

Reply:
I think we prefer more readable original code.

Reply:
Added commit with extra comments for clarity.

Reply:
Rather than extra comments, please convert to using an `IEnumerable` form

Reply:
This pattern can be found in https://github.com/dotnet/runtime/blob/f7e0069ef27afa83c38d97c1650bc87e152d98a2/src/libraries/System.Linq/src/System/Linq/Single.cs#L31

Reply:
I have extracted the code to a new `TrySingle` method, which is already proposed for LINQ: https://github.com/dotnet/runtime/issues/16362